ofrak_core/test_ofrak/unit/test_ofrak_context.py

```diff
@@ -14,6 +14,32 @@
 
 from ofrak.core.filesystem import FilesystemRoot
 
+from ofrak.model.component_model import ComponentExternalTool
+from ofrak.core.squashfs import _UnsquashfsV45Tool
+from ofrak.core.strings_analyzer import _StringsToolDependency
+
+
+def test_path_auth_error(monkeypatch, caplog):
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+
+    if os.name != "posix":
+        LOGGER.warning("test_path_auth_error is currently only supported by POSIX systems.")
+        return
+
+    monkeypatch.setenv("PATH", "/root")
+
+    for tool in (
+        ComponentExternalTool("apktool", "", "-version"),
+        _UnsquashfsV45Tool(),
+        _StringsToolDependency(),
+    ):
+        task = loop.create_task(tool.is_tool_installed())
+        loop.run_until_complete(task)
+        expected = f"Encountered PermissionError while searching PATH for {tool.tool}."
+        found = [r for r in caplog.records if r.message == expected]
+        assert len(found) == 1, "Missing error message from component_model"
```

Comment:
```suggestion
        assert any(r.message == expected for r in caplog.records), "Missing error message from component_model"
```

This way of doing it is a little more concise. Will probably need to re-run the linter after applying the suggestion, though.

Reply:
On a second read, my suggested change is not exactly the same as what the original test does, so feel free to resolve this. 

But this particular test is failing in the CI check, so this will have to be revised in some way before we can merge.